File: Go/1768. Merge Strings Alternately.go
package main

//goland:noinspection GoUnusedFunction
func mergeAlternately(word1 string, word2 string) string {
	var result []byte

	length := max(len(word1), len(word2))

	for i := 0; i < length; i++ {
		if i < len(word1) {
			result = append(result, word1[i])
		}

		if i < len(word2) {
			result = append(result, word2[i])
		}
	}
	return string(result)
}

File: JavaScript/1768. Merge Strings Alternately.js
// noinspection JSUnusedGlobalSymbols
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
const mergeAlternately = (word1, word2) => {
  const length = Math.max(word1.length, word2.length)
  let result = ''

  for (let i = 0; i < length; i++) {
    result += (word1[i] || '') + (word2[i] || '')
  }

  return result
}
